ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml

```diff
@@ -0,0 +1,111 @@
+---
+
+- name: Update Package Cache (apt/Ubuntu)
+  apt:
+    update_cache: yes
+  changed_when: false
+  when: ansible_distribution == "Ubuntu"
+  become: yes
+
+- name: Check Nagios Installation
+  stat:
+    path: /usr/local/nagios/etc/nagios.cfg
+  register: nagios_installation
+
+- name: Install dependencies for building Nagios Core {{ nagios_version }}
+  package:
+    name: "{{ pkg_name }}"
+    state: latest
+  become: yes
+  loop: "{{ pkg_list }}"
+  loop_control:
+    loop_var: pkg_name
+  when: not nagios_installation.stat.exists
+
+- name: Download + Unpack Nagios Core {{ nagios_version }} source
+  unarchive:
+    src: "https://assets.nagios.com/downloads/nagioscore/releases/nagios-{{ nagios_version }}.tar.gz"
+    remote_src: yes
+    dest: "/tmp"
+  when: not nagios_installation.stat.exists
+
+# 'Creates' argument keeps the playbook idempotent.
+- name: Run './configure' on source.
+  command:
+    cmd: "./configure {{ extra_configure_args }}"
+    chdir: "/tmp/nagios-{{ nagios_version }}/"
+    creates: "/tmp/nagios-{{ nagios_version }}/Makefile"
```

Comment:
do we need to have this pre-check for existence of Makefile? assume nagios_installation is not installed on this machine then does not really matter if the same version nagios already exists in /tmp